src/main/java/com/ecore/roles/model/Membership.java

```diff
@@ -26,14 +23,17 @@ public class Membership {
     @Type(type = "uuid-char")
     private UUID id;
 
-    @OneToOne
+    @ManyToOne
     @JoinColumn(name = "role_id", nullable = false)
+    @NotNull(message = "Role is required")
```

Comment:
I think it makes more sense to use Bean Validation instead of throwing InvalidArgumentException. The code becomes cleaner with its annotations: @NotNull, @NotEmpty, etc...